### ISSUE TITLE
SMS_saveROMBank() implementation in SMSLib.h makes it only usable in the same scope as SMS_restoreBank()

### ISSUE THREAD

**Issue by kusfo** [sverx/devkitSMS#53]
SMS_saveROMBank is implemented as the following define: 

`#define SMS_saveROMBank()       unsigned char _saved_slot2_ROM_bank = ROM_bank_to_be_mapped_on_slot2`

This causes the var _saved_slot2_ROM_bank to be defined only in the same scope where SMS_saveROMBank is used. So, any call to SMS_restoreROMBank outside that scope will fail. 

As an example, in my codebase have something like
`if(flashCartridge) {
     SMS_saveROMBank();
`

**Comment by kusfo** [comment]
Yes, for my case I just put them together, but it's a bug that may not be not so obvious to the newbie :-)

**Comment by sverx** [comment]
Well, not really a bug in my opinion but surely it's an issue some other might stumble upon.
Maybe the solution is just to use a global variable instead of a local one? Also, what do you think about this @raphnet ? :thinking:

**Comment by sverx** [comment]
OK so maybe I could add a function that returns the currently mapped ROM bank value (a byte).

This way, one will be able to declare his own variable with the desired scope (either global or local to a function) and save the currently mapped ROM bank there.

Basically something like this would probably just work:
`#define SMS_getROMBank()       (ROM_bank_to_be_mapped_on_slot2)`

I'm not sure about the name though. It needs to be clear that you're reading back the number of the currently mapped ROM bank... :thinking: 

Suggestions?

**Comment by sverx** [comment]
No, I'm not going to remove the existing save and restore features, just adding some new ones.
